### DISCUSSION TITLE
Does AWS Advanced JDBC Wrapper Support Failover Across AZs within the Same Region for Global Database?

### DISCUSSION THREAD

**Discussion by cprayer** [aws/aws-advanced-jdbc-wrapper#868] [Q&A] [answered]
Hi, I am currently using RDS in my company. 
I want to change my RDS cluster from a regional cluster to a global database. I am currently using aws-advanced-jdbc-wrapper in my JVM application. 
Looking at the README, it says that global database is not supported, but I was wondering if inter-AZ failover is also not supported or only inter-region failover is not supported.

Thank you for your time and assistance.

Best regards,

**Comment by aaronchung-bitquill** [comment]
Hi @cprayer

Thank you for reaching out. 

Unfortunately, since the AWS JDBC Driver does not currently support Global Databases, that also includes failover across AZs within the same region for Global Databases. 
Support for Global Databases is in the backlog but we cannot comment on a timeline right now. 

If you have any other questions or concerns, please feel free reach out again. 
Thank you

**Reply by cprayer** [reply]
@aaronchung-bitquill Thank you for your response. 

So, if I need to failover between AZs within a region, can I just use domain-based failover (I don't need to modify the cluster endpoint in my application)?
If I need inter-region failover, can I just change the cluster endpoint in the application or use the write forwarding feature?

**Reply by cprayer** [reply]
@aaronchung-bitquill 

Just bumping up my previous question in case you missed it. Would really appreciate your insights when you have a moment. Thanks!

**Reply by aaronchung-bitquill** [reply]
Hi @cprayer 

Apologies for missing your question. 

For clarification, for both questions, would you still be using Global Databases?

**Reply by cprayer** [reply]
Hello~ @aaronchung-bitquill 

I saw it late due to the holiday period in Korea.
Yes, we are currently using only the global database role. We are not using regional cluster role.

![image](https://github.com/awslabs/aws-advanced-jdbc-wrapper/assets/24709996/f57568fe-856b-4f2c-bd10-6431edb818ea)

**Reply by aaronchung-bitquill** [reply]
Hi @cprayer 

If you plan on using the AWS JDBC Driver with Global Databases, at this time I unable to recommend that. 
Since the AWS JDBC Driver does not currently support Global Databases, there may be other unforeseen errors/issues that we do not know about yet besides the issues with failing over to a secondary cluster.

**Reply by cprayer** [reply]
@aaronchung-bitquill 
It is recommended not to use aws-advanced-jdbc-wrapper for either failover within an AZ or between regions, right?

**Reply by aaronchung-bitquill** [reply, marked as answer]
@cprayer If using Global Databases, then yes it is currently not recommended to use aws-advajced-jdbc-wrapper for either failover within an AZ or between regions.

**Reply by cprayer** [reply]
Thank you for your help.
